### PR TITLE
support mattermost integration

### PR DESCRIPTION
**Pull request by ryota-sakamoto**
Signed-off-by: Ryota Sakamoto <sakamo.ryota+github@gmail.com>

refs: #176

I don't use [mattermost/mattermost-server model/client4.go](https://github.com/mattermost/mattermost-server/blob/master/model/client4.go), because it is too large to use library.

### CODE REVIEW

**Review by alexmt — APPROVED** [review]
LGTM!

File: pkg/services/mattermost.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	"io/ioutil"
+	"net/http"
+	texttemplate "text/template"
+
+	log "github.com/sirupsen/logrus"
+
+	httputil "github.com/argoproj-labs/argocd-notifications/pkg/util/http"
+)
+
+type MattermostNotification struct {
```
Comment (alexmt) [inline pkg/services/mattermost.go:16]: I was under impression that we can just re-use Slack notification and templates but looks like not every field is supported. 

I see attachments in documentation but no blocks, users etc https://docs.mattermost.com/developer/message-attachments.html#attachment-options

@ryota-sakamoto can you please confirm that I understand it correctly?

Comment (alexmt) [reply pkg/services/mattermost.go:16]: Basically mattermost and slack are similar but not really compatible, right?

Comment (ryota-sakamoto) [reply pkg/services/mattermost.go:16]: @alexmt 
yes. I cannot find it out that block is supported.
I guess it is only similar for attachments.
https://mattermost.atlassian.net/browse/MM-26729

**Pull request merged**